File: Programmers/C++/땅따먹기.cpp
#include <iostream>
#include <vector>
#define FOR(i,n) for(int i=0;i<(n);++i)
#define FORS(i,s,n) for(int i=(s);i<(n);++i)
using namespace std;

int res,mx;

int solution(vector<vector<int> > land) {
    FORS(y,1,land.size()) {
        FOR(x,land[0].size()) {
            int mx = 0;
            FOR(px,land[0].size()) {
                if(x==px) continue;
                mx = max(mx,land[y-1][px]);
            }
            land[y][x]+=mx;
        }
    }

    FOR(x,land[0].size()) res=max(res,land[land.size()-1][x]);
    return res;
}

File: Programmers/Swift/땅따먹기.swift
import Foundation

func solution(_ land:[[Int]]) -> Int{
    var mx = 0

    var mp = land;
    for y in 1..<mp.count {
        for x in 0..<mp[0].count {
            var mx = 0
            for px in 0..<mp[0].count {
                if x == px {continue;}
                mx = max(mx,mp[y-1][px])
            }
            mp[y][x] += mx
        }
    }
    var res = 0
    for x in 0..<mp[0].count {
        res = max(res, mp[mp.count-1][x])
    }
    return res
}
